src/main/java/com/ibm/crail/storage/blkdev/BlkDevStorageServer.java

```diff
@@ -57,27 +59,41 @@ public void init(CrailConfiguration crailConfiguration, String[] args) throws Ex
 			throw new IOException("BlkDevStorageTier already initialized");
 		}
 		initialized = true;
+
 		BlkDevStorageConstants.init(crailConfiguration, args);
-		LOG.info("initalizing block device datanode");
-		// We do not support multiple block devices yet
-		InetAddress address = InetAddress.getLoopbackAddress();
-		int port = 12345;
-
-		storageAddr = new InetSocketAddress(address, port);
-		String directory = BlkDevStorageConstants.DATA_PATH;
-		path = FileSystems.getDefault().getPath(directory);
-		if (!Files.exists(path)) {
-			throw new IllegalArgumentException("BlkDev path does not exists!");
-		}
+
+		storageAddr = getDataNodeAddress();
 		isAlive = true;
-		alignedSize = BlkDevStorageConstants.STORAGE_LIMIT - (BlkDevStorageConstants.STORAGE_LIMIT % BlkDevStorageConstants.ALLOCATION_SIZE);
-		endPoint = new BlkDevStorageEndpoint();
+		alignedSize = BlkDevStorageConstants.STORAGE_SIZE -
+				(BlkDevStorageConstants.STORAGE_SIZE % BlkDevStorageConstants.ALLOCATION_SIZE);
 		addr = 0;
 	}
 
+	public static InetSocketAddress getDataNodeAddress() throws IOException {
+		String ifname = BlkDevStorageConstants.STORAGE_BLKDEV_INTERFACE;
```

Comment:
I don't recommend choosing the IP via the interface. I know that we do it like this with the RDMA datanode but the interface can have multiple IPs i.e. it does not uniquely identify an IP address. Introduce an IP command line parameter which can be set when starting the storage tier.

Reply:
Ok. I think it makes sensethen to have a constant in which I also specify the IP address on the target side:
String ipAddress = BlkDevStorageConstants.STORAGE_BLKDEV_IPADDRESS;